pkg/resources/schema.go

```diff
@@ -29,268 +33,380 @@ var schemaSchema = map[string]*schema.Schema{
 	database := d.Get("database").(string)
-
-	ctx := context.Background()
-
-	createReq := &sdk.CreateSchemaOptions{
-		Transient:         GetPropertyAsPointer[bool](d, "is_transient"),
-		WithManagedAccess: GetPropertyAsPointer[bool](d, "is_managed"),
-		Tag:               getPropertyTags(d, "tag"),
-		Comment:           GetPropertyAsPointer[string](d, "comment"),
+	id := sdk.NewDatabaseObjectIdentifier(database, name)
+
+	dataRetentionTimeInDays,
+		maxDataExtensionTimeInDays,
+		externalVolume,
+		catalog,
+		replaceInvalidCharacters,
+		defaultDDLCollation,
+		storageSerializationPolicy,
+		logLevel,
+		traceLevel,
+		suspendTaskAfterNumFailures,
+		taskAutoRetryAttempts,
+		userTaskManagedInitialWarehouseSize,
+		userTaskTimeoutMs,
+		userTaskMinimumTriggerIntervalInSeconds,
+		quotedIdentifiersIgnoreCase,
+		enableConsoleOutput,
+		err := GetAllDatabaseParameters(d)
+	if err != nil {
+		return diag.FromErr(err)
 	}
 
-	dataRetentionTimeInDays := GetPropertyAsPointer[int](d, "data_retention_days")
-	if dataRetentionTimeInDays != nil && *dataRetentionTimeInDays != IntDefault {
-		createReq.DataRetentionTimeInDays = dataRetentionTimeInDays
+	opts := &sdk.CreateSchemaOptions{
+		Transient:                               GetConfigPropertyAsPointerAllowingZeroValue[bool](d, "is_transient"),
+		WithManagedAccess:                       GetConfigPropertyAsPointerAllowingZeroValue[bool](d, "with_managed_access"),
+		DataRetentionTimeInDays:                 dataRetentionTimeInDays,
+		MaxDataExtensionTimeInDays:              maxDataExtensionTimeInDays,
+		ExternalVolume:                          externalVolume,
+		Catalog:                                 catalog,
+		ReplaceInvalidCharacters:                replaceInvalidCharacters,
+		DefaultDDLCollation:                     defaultDDLCollation,
+		StorageSerializationPolicy:              storageSerializationPolicy,
+		LogLevel:                                logLevel,
+		TraceLevel:                              traceLevel,
+		SuspendTaskAfterNumFailures:             suspendTaskAfterNumFailures,
+		TaskAutoRetryAttempts:                   taskAutoRetryAttempts,
+		UserTaskManagedInitialWarehouseSize:     userTaskManagedInitialWarehouseSize,
+		UserTaskTimeoutMs:                       userTaskTimeoutMs,
+		UserTaskMinimumTriggerIntervalInSeconds: userTaskMinimumTriggerIntervalInSeconds,
+		QuotedIdentifiersIgnoreCase:             quotedIdentifiersIgnoreCase,
+		EnableConsoleOutput:                     enableConsoleOutput,
+		PipeExecutionPaused:                     GetConfigPropertyAsPointerAllowingZeroValue[bool](d, "pipe_execution_paused"),
+		Comment:                                 GetConfigPropertyAsPointerAllowingZeroValue[string](d, "comment"),
 	}
-
-	err := client.Schemas.Create(ctx, sdk.NewDatabaseObjectIdentifier(database, name), createReq)
-	if err != nil {
-		return fmt.Errorf("error creating schema %v err = %w", name, err)
+	if strings.EqualFold(strings.TrimSpace(name), "PUBLIC") {
```

Comment:
I think there's also a schema named local mentioned [here](https://docs.snowflake.com/en/sql-reference/sql/show-schemas#usage-notes), but I'm not sure if you can `alter` or `create or alter` it.

Reply:
It behaves like a normal schema and doesn't require any specific handling.